app/assets/javascripts/show_errors.js

```diff
@@ -37,14 +37,23 @@ let ErrorDisplay = {
   hideError(e) {
     $(e.target).removeClass('has-error').parent().removeClass('has-error');
     $(e.target).siblings('.error-msg').remove();
+    $(e.target).parent('.error-msg').remove();
   },
 
   findField(field_name, $form, response) {
     if (response.name) {
       field_name = [response.name, '[', field_name, ']'].join('');
     }
-    return $form.find("[name='" + field_name + "']");
+    var field = $form.find("[name='" + field_name + "']");
+    if (!field.exists()) {
+      field = $form.find(':submit').prev();
+    }
+    return field
   },
 }
 
+$.fn.exists = function () {
+    return this.length !== 0;
```

Comment:
it kinda makes it less readable for me cause it's not the standard way of doing it, so it makes me start wondering what's going on. if you like it we can keep it but I would like the monkey patch to be in it's own file.